src/components/Pages/SetupPage/SetupPage.tsx

```diff
@@ -47,20 +123,77 @@ export const SetupPage = () => {
       setNameError('')
     }
 
+    // Validate the email.
     if (email && !email.includes('@')) {
       setEmailError('Please enter a valid email.')
     } else {
       setEmailError('')
     }
 
+    // Validate the agree checkbox.
     if (!agree) {
       setAgreeError('Please accept the terms of use and privacy policy.')
     } else {
       setAgreeError('')
     }
   }, [name, email, agree])
 
-  if (!initialized) {
+  // Initialization effect.
+  // Will run some checks to see if the user can proceed with the simplified avatar setup flow.
+  useEffect(() => {
+    ;(async () => {
+      const toLogin = () => {
+        window.location.href = '/auth/login'
+      }
+
+      // Check if the wallet is connected.
+      try {
+        await connection.tryPreviousConnection()
+      } catch (e) {
+        console.warn('No previous connection found')
+        toLogin()
+        return
+      }
+
+      const provider = await connection.getProvider()
+      const accounts = (await provider.request({ method: 'eth_accounts' })) as string[]
+
+      // Check that there is at least one account connected.
+      if (!accounts.length) {
+        console.warn('No accounts found')
+        toLogin()
+        return
+      }
+
+      const account = accounts[0]
+
+      accountRef.current = account
+
+      const profile = await fetchProfile(account)
+
+      // Check that the connected account does not have a profile already.
+      if (profile) {
```

Comment:
If the profile already exist, what do you think about skipping the setup and going to the `redirectTo` site?

Reply:
Done